packages/beacon-node/src/chain/seenCache/seenGossipBlockInput.ts

```diff
@@ -371,36 +371,36 @@ export class SeenGossipBlockInput {
 
   private async reconstructColumns(config: ChainForkConfig, blockRoot: RootHex) {
     const blockCache = this.blockInputCache.get(blockRoot);
-    if (blockCache === undefined || blockCache.fork !== ForkName.fulu) {
+    if (
+      blockCache === undefined ||
+      blockCache.fork !== ForkName.fulu ||
+      blockCache.cachedData === undefined ||
+      blockCache.cachedData.fork !== ForkName.fulu
+    ) {
       return;
     }
 
     // If already have all columns, exit
-    if (
-      blockCache.cachedData &&
-      (blockCache.cachedData.fork !== ForkName.fulu ||
-        this.hasSampledDataColumns(blockCache.cachedData.dataColumnsCache))
-    ) {
+    if (this.hasSampledDataColumns(blockCache.cachedData.dataColumnsCache)) {
       return;
     }
 
-    // Process KZG commitments into versioned hashes
-    let versionedHashes: Uint8Array[];
-
+    let commitments: undefined | Uint8Array[];
     if (blockCache.block) {
       const block = blockCache.block as fulu.SignedBeaconBlock;
-      versionedHashes = block.message.body.blobKzgCommitments.map(kzgCommitmentToVersionedHash);
-    } else if (blockCache.cachedData) {
+      commitments = block.message.body.blobKzgCommitments;
+    } else {
       const firstSidecar = blockCache.cachedData.dataColumnsCache.values().next().value;
```

Comment:
I think there needs to be a nullish check here if you don't have the `else if`, right?
```suggestion
      const firstSidecar = blockCache.cachedData?.dataColumnsCache.values().next().value;
```